Frontiers_186_list_of_customer_emails.py

```diff
@@ -0,0 +1,40 @@
+"""
+ Write a py script that will get list of all customer emails for our website
+Setup:
+You need the ecom website.
+You need API credentials.
+
+Problem statement:
+Write a script that will get a list of all users for the website.
+Your script should output a csv file with list of email addresses.
+"""
+
+from woocommerce import API
+import csv
+
+wcapi = API(
+    url = "http://localhost:8888/mysite2",
+    consumer_key = "<add consumer key>",
+    consumer_secret = "<add consumer secret>",
+    version = "wc/v3"
+)
+
+page = 1
+email_list = []
+while True:
+    r = wcapi.get("customers", params={"per_page": 100, "page": page})
+    customers = r.json()
+    output_file = 'list_of_user_email.csv'
+
+    page += 1
+    if not customers:
+        break
+
+    for customer in customers:
+        customer_email = customer["email"]
+        email_list.append([customer_email])
```

Comment:
Why are you adding alist to the list of email. Why not add the email. Now you have a list of lists, not a list of strings.
I would' ven done email_list.append(customer_email)
But I think you writer.writerows() might change too if you make this change.

Reply:
So I used chat gpt to help me with this. As it is if I change it to email_list.append(customer_email) the writer.writerows() will write each character separated by a comma.  to get around this I used a for loop to split each item at the comma (thanks again chat gpt)